src/script/lua_api/l_util.cpp

```diff
@@ -238,14 +238,29 @@ int ModApiUtil::l_check_password_entry(lua_State *L)
 	return 1;
 }
 
-// get_password_hash(name, raw_password)
+// get_password_hash(name, password)
 int ModApiUtil::l_get_password_hash(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
+
+	// only accept strings, not numbers
+	// we do strong type checking here because mods have misused the API in the past.
+	luaL_checktype(L, 1, LUA_TSTRING);
 	std::string name = luaL_checkstring(L, 1);
-	std::string raw_password = luaL_checkstring(L, 2);
-	std::string hash = translate_password(name, raw_password);
-	lua_pushstring(L, hash.c_str());
+
+	luaL_argcheck(L, !name.empty(), 1,
+		"User name must be non-empty, use minetest.sha1 or minetest.sha256 if you need a generic hash function.\n"
+		"See documentation for more details on how to migrate legacy code.");
+
+	luaL_checktype(L, 2, LUA_TSTRING);
+	std::string password = luaL_checkstring(L, 2);
```

Comment:
should be `checkParam<std::string>(L, 2)` unless we're intentionally not allowing null bytes

Reply:
it wasn't intentional but now that you mention it, I don't see any reason to allow null bytes. `check_password_entry` won't allow them and neither would a client.
Actually should probably modify the example in the documentation to reflect that the old implementation didn't allow null bytes.

Reply:
should probably check for unprintable characters in general

Reply:
that would be too much IMO. just don't support null bytes

Reply:
maybe you're right, tried implementing it but having issues with locales. Might just check for null characters.

Reply:
I wouldn't even do that, just use `luaL_checkstring` like before